### ISSUE TITLE
Loss function ignore_index

### ISSUE THREAD

**Issue by aashish33** [pytorch/pytorch#30984]
I have a simple question. I want to confirm that when I specify a class index in ignore_index, and reduction = 'mean', then the loss is not averaged over the ignored class.

In the documentation, it says "When size_average is True, the loss is averaged over non-ignored targets." Just want to make sure that it will behave the same when reduction is 'mean'
